libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1524,18 +1524,48 @@ bool ExpressionCompiler::visit(IndexAccess const& _indexAccess)
 				setLValueToStorageItem(_indexAccess);
 			break;
 		case DataLocation::Memory:
+			ArrayUtils(m_context).accessIndex(arrayType);
 			setLValue<MemoryItem>(_indexAccess, *_indexAccess.annotation().type, !arrayType.isByteArray());
 			break;
 		case DataLocation::CallData:
-			//@todo if we implement this, the value in calldata has to be added to the base offset
```

Comment:
Git blame shows this is from `2015-06-25 17:51:01` (nb: pre mainnet launch) - it feels sooo good to finally fix this!